core/templates/pages/exploration-editor-page/services/exploration-correctness-feedback.service.ts

```diff
@@ -17,27 +17,49 @@
  * exploration.
  */
 
-require(
-  'pages/exploration-editor-page/services/exploration-property.service.ts');
+import { ExplorationPropertyService } from 'pages/exploration-editor-page/services/exploration-property.service';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { AlertsService } from 'services/alerts.service';
+import { ChangeListService } from './change-list.service';
+import { LoggerService } from 'services/contextual/logger.service';
 
-angular.module('oppia').factory('ExplorationCorrectnessFeedbackService', [
-  'ExplorationPropertyService', function(ExplorationPropertyService) {
-    var child = Object.create(ExplorationPropertyService);
-    child.propertyName = 'correctness_feedback_enabled';
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationCorrectnessFeedbackService extends
+  ExplorationPropertyService {
+  correctnessFeedbackIsEnabled: boolean = false;
 
-    child._isValid = function(value) {
-      return (typeof value === 'boolean');
-    };
+  constructor(
+    protected alertsService: AlertsService,
+    protected changeListService: ChangeListService,
+    protected loggerService: LoggerService
+  ) {
+    super(alertsService, changeListService, loggerService);
+  }
+
+  propertyName: string = 'correctness_feedback_enabled';
 
-    child.isEnabled = function() {
-      return child.savedMemento;
-    };
+  isEnabled(): boolean {
+    return Boolean(this.savedMemento);
+  }
 
-    child.toggleCorrectnessFeedback = function() {
-      child.displayed = !child.displayed;
-      child.saveDisplayedValue();
-    };
+  toggleCorrectnessFeedback(): void {
+    // Earlier we were using this.displayed = !this.displayed this is wrong.
+    // Because we the variable displayed is string type and we were storing
+    // boolean init by using !, To overcome this problem JSON.parse('true')
+    // and correctnessFeedbackIsEnabled is used.
 
-    return child;
+    this.correctnessFeedbackIsEnabled = !this.correctnessFeedbackIsEnabled;
+    if (this.correctnessFeedbackIsEnabled) {
+      this.displayed = JSON.parse('true');
```

Comment:
Can you explain, how this works?

`JSON.parse('true')` just evaluates to boolean `true`.

Reply:
+1: I'm wondering the same thing!